client/src/api/datasource/__tests__/ApiClient.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { test } from 'vitest';
+import nock from 'nock';
+import { ApiClient } from '../ApiClient';
+import { DataSource } from '../../Models';
+
+test('list should return a list of data sources', async ({ expect }) => {
+  const data: DataSource[] = [
+    {
+      type: 'API',
+      name: 'gnuradio',
+      account: 'gnuradio',
+      container: 'iqengine',
+      description: 'GNU Radio hosted test data',
+    },
+    {
+      type: 'API',
+      name: 'another',
+      account: 'another',
+      container: 'iqengine',
+      description: 'Another GNU Radio hosted test data',
+    },
+  ];
+  nock('http://localhost:3000').get('/api/datasources').reply(200, data);
+
+  const client = new ApiClient();
+  const result = await client.list();
+  expect(result).toEqual(data);
+});
+
+test('get should return a data source', async ({ expect }) => {
+  const data: DataSource = {
+    type: 'API',
+    name: 'gnuradio',
+    account: 'gnuradio',
+    container: 'iqengine',
+    description: 'GNU Radio hosted test data',
+  };
+  nock('http://localhost:3000').get('/api/datasources/gnuradio/iqengine').reply(200, data);
+
+  const client = new ApiClient();
+  const result = await client.get('gnuradio', 'iqengine');
+  expect(result).toEqual(data);
+});
```

Comment:
I would add a describe as a way to group those tests so we know they are part of the same class that we are testing.